astacus/coordinator/coordinator.py

```diff
@@ -111,15 +111,17 @@ def delete_json(self, name: str) -> None:
         finally:
             self.state.cached_list_response = None
 
-    def download_json(self, name: str) -> Json:
-        return self.storage.download_json(name)
+    @contextlib.contextmanager
+    def open_json_bytes(self, name: str) -> Iterator[bytearray]:
+        with self.storage.open_json_bytes(name) as json_bytes:
+            yield json_bytes
 
     def list_jsons(self) -> list[str]:
         return self.storage.list_jsons()
 
-    def upload_json_str(self, name: str, data: str) -> bool:
+    def upload_json_bytes(self, name: str, data: bytes | bytearray) -> bool:
```

Comment:
note `bytearray` is a subclass of `bytes`.


```suggestion
    def upload_json_bytes(self, name: str, data: bytes) -> bool:
```

Reply:
```>>> issubclass(bytearray, bytes)
False
```
?

Reply:
I mean in terms of mypy.  subclass is the wrong word but the type `bytearray` is accepted everywhere `bytes` is and you can write stuff like this

```python
test: bytes = bytearray()
```

Reply:
Subtype is the word.

Reply:
This doesn't apply anymore with things typed with mmap.